src/contents/unrrugable-ai-agents.mdx

```diff
@@ -0,0 +1,219 @@
+
+```python
+# Check if notebook is in Colab
+try:
+    # Install ezkl and onnx if in Colab
+    import google.colab
+    import subprocess
+    import sys
+    subprocess.check_call([sys.executable, "-m", "pip", "install", "ezkl"])
+    subprocess.check_call([sys.executable, "-m", "pip", "install", "onnx"])
+except:
+    pass  # Rely on local installation of ezkl otherwise
+
+# Import required libraries
+from torch import nn
+import torch
+import ezkl
+
+# Define the address that will receive the tip
+tip_recipient_address = 0x707e55a12557E89915D121932F83dEeEf09E5d70
+
+# Convert the address to the format expected by EZKL
+def address_to_byte_tensor(address):
+    address_hex = f"{address:040x}"
+    return torch.tensor([int(address_hex[i:i+2], 16) for i in range(0, len(address_hex), 2)], dtype=torch.uint8)
+
+tip_recipient_tensor = address_to_byte_tensor(tip_recipient_address)
+print(f"Tip Recipient Tensor: {tip_recipient_tensor}")
+
+# Define a model that outputs the tipped address
+class FixedHexAddressModel(nn.Module):
+    def __init__(self, address_tensor):
+        super(FixedHexAddressModel, self).__init__()
+        self.output_value = address_tensor
+
+    def forward(self, x):
+        # Ignore input and always return the fixed address
+        batch_size = x.size(0)
+        return self.output_value.repeat(batch_size, 1)  # Repeat for batch size
+
+# Instantiate the model
+circuit = FixedHexAddressModel(tip_recipient_tensor)
+
+# Example input
+dummy_input = torch.ones(1, 1)  # Dummy input (not used by the model)
+
+# Generate the output
+output = circuit(dummy_input)
+print(f"Output (Hex Address): {output}")
+
+# Export the model to ONNX format
+torch.onnx.export(
+    circuit,
+    dummy_input,
+    "fixed_hex_address_model.onnx",
+    input_names=["input"],
+    output_names=["output"],
+    dynamic_axes={"input": {0: "batch_size"}, "output": {0: "batch_size"}}
+)
+
+print("Model saved as 'fixed_hex_address_model.onnx'!")
+```
+
+## Step 1: Create a Verifier Contract
+
+Next, run all the other code snippets in Google Collab, this will generate a ZK circuit that is verifiable off-chain. Now we want to be able to verify on-chain, to do it run the following snippet. This will print a fully working Solidity verifier contract.
+
+```python
+abi_path = 'test.abi'
+sol_code_path = 'test.sol'
+
+res = await ezkl.create_evm_verifier(
+        vk_path,
+        
+        settings_path,
+        sol_code_path,
+        abi_path,
+    )
+
+assert res == True
+
+with open(sol_code_path, 'r') as sol_file:
+    sol_content = sol_file.read()
+    print(sol_content)
+```
+
+The next step is deploying to your preferred chain. Please note that compiler optimizations must be enabled to successfully compile the contract. This is necessary because the generated contract is quite large and requires optimization to fit within deployment constraints.
```

Comment:
Suggestion: would it be a good idea to bold "Please note..."? In a way, highlighting the importance of optimizing amidst this text body

Reply:
Yes, good idea